### DISCUSSION TITLE
Custom parser in browser

### DISCUSSION THREAD

**Discussion by makssein** [socketio/socket.io#4854] [Q&A] [answered]
How to use a custom parser if the client is a browser?

My server code:
```
io = new Server(HTTPServer, {
            wsEngine: eiows.Server,
            cors: {
                origin: process.env.SOCKET_ORIGIN,
                methods: ["GET", "POST"],
                credentials: true,
                transports: ['websocket', 'polling']
            },
            allowEIO3: true,
            parser:  require("socket.io-msgpack-parser")
        });
```

Client: 
```
<head>
...
<scrpit src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.7.1/socket.io.msgpack.min.js"></scrpit>

    <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.7.1/socket.io.js" integrity="sha512-Z6C1p1NIexPj5MsVUunW4pg7uMX6/TT3CUVldmjXx2kpip1eZcrAnxIusDxyFIikyM9A61zOVNgvLr/TGudOQg==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
...
</head>

...

socket = io(socketEndPoint, {
        withCredentials: true,
});
```
All I get is endless connection attempts (https://i.imgur.com/e418I5s.png)

**Comment by darrachequesne** [comment, marked as answer]
Hmm... That's weird, this should be sufficient.

I wasn't able to reproduce: https://github.com/socketio/socket.io-fiddle/tree/socket.io/discussions/4854

**Reply by makssein** [reply]
I decided to display the reason for the disconnection: on the client - “transport close”, and on the server the disconnect event does not even fire

**Reply by makssein** [reply]
@darrachequesne 
That's my fault. In the head of the layout, I added the regular library and the socket.io.msgpack.min.js library at the same time.
I thought that the second was just an addition to the first.
